rust/protocol/src/session_cipher.rs

```diff
@@ -296,21 +312,53 @@ fn decrypt_message_with_record<R: Rng + CryptoRng>(
     ciphertext: &SignalMessage,
     csprng: &mut R,
 ) -> Result<Vec<u8>> {
+    let log_decryption_failure = |state: &SessionState, error: &SignalProtocolError| {
+        log::error!(
+            "Failed to decrypt whisper message with ratchet key: {} and counter: {}. \
+             Session loaded for {}. Local session has base key: {} and counter: {}. {}",
+            hex::encode(ciphertext.sender_ratchet_key().public_key_bytes().unwrap()),
```

Comment:
Is this `unwrap` really safe?

Might be worth adding a util that hex encodes an `Optional<AsRef<[u8]>>` that returns something like "{missing}" when `None`?